### ISSUE TITLE
Handle validation errors in api handlers

### ISSUE THREAD

**Issue by 3coins** [jupyter-server/jupyter-scheduler#306]
## Description
Passing invalid values in the query/payload to the API handlers causes the handler to fail and returns the complete traceback to the  UI.

## Reproduce
Call the `list_jobs` api with `max_items` as a non-numeric value, for example

**Request**
```
/scheduler/jobs?max_items=abc
```

**Response**
```
{
  "message": "Unhandled error",
  "reason": null,
  "traceback": "Traceback (most recent call last):\n  File \"/opt/anaconda3/envs/jupyter-scheduler/lib/python3.9/site-packages/tornado/web.py\", line 1713, in _execute\n    result = await result\n  File \"/Users/pijain/projects/jupyter-scheduler/jupyter-scheduler/jupyter_scheduler/handlers.py\", line 172, in get\n    list_jobs_query = ListJobsQuery(\n  File \"pydantic/main.py\", line 342, in pydantic.main.BaseModel.__init__\npydantic.error_wrappers.ValidationError: 1 validation error for ListJobsQuery\nmax_items\n  value is not a valid integer (type=type_error.integer)\n"
}
```

## Expected behavior

A human readable error should be returned without the full traceback value.

## Context
Version: 1.1.4

**Comment by jweill-aws** [comment]
Should we have a "debug" mode that displays the full traceback (e.g., for diagnosing errors during development) and that is disabled by default?

**Comment by edwardps** [comment]
Is this fix included by 1.1.4? I still saw the same issue with 1.1.4.

```
curl "https://xxxx/jupyter/default/scheduler/jobs?max_items=25'&1671483042203" .....
{"message": "Unhandled error", "reason": null, "traceback": "Traceback (most recent call last):\n  File \"/opt/conda/envs/studio/lib/python3.9/site-packages/tornado/web.py\", line 1704, in _execute\n    result = await result\n  File \"/opt/conda/envs/studio/lib/python3.9/site-packages/jupyter_scheduler/handlers.py\", line 172, in get\n    list_jobs_query = ListJobsQuery(\n  File \"pydantic/main.py\", line 342, in pydantic.main.BaseModel.__init__\npydantic.error_wrappers.ValidationError: 1 validation error for ListJobsQuery\nmax_items\n  value is not a valid integer (type=type_error.integer)\n"}
```
